cedar-policy-core/src/ast/request.rs

```diff
@@ -382,6 +379,21 @@ pub enum ContextCreationError {
     ExprConstruction(#[from] ExprConstructionError),
 }
 
+impl ContextCreationError {
+    pub(crate) fn not_a_record(expr: RestrictedExpr) -> Self {
+        Self::NotARecord(NotARecord {
+            expr: Box::new(expr),
+        })
+    }
+}
+
+/// Error type for an expression that is not a record
+#[derive(Debug)]
+pub struct NotARecord {
+    /// Expression which is not a record
+    expr: Box<RestrictedExpr>,
+}
```

Comment:
I know I'm late and this has already been merged, but since this is an error type, it would probably make sense for it to derive `Error`, and have the error message defined here (and `ContextCreationError::NotARecord` just have `error(transparent)`)